lib/api/src/instance.rs

```diff
@@ -85,11 +124,21 @@ impl Instance {
             })
             .collect::<Exports>();
 
-        Ok(Self {
+        let instance = Self {
             handle: Arc::new(Mutex::new(handle)),
             module: module.clone(),
             exports,
-        })
+        };
+
+        unsafe {
```

Comment:
I'm not super fan of this function.
I'm thinking on ways we can do it differently, but haven't got any so far.

Can you add a block comment before the unsafe function commenting what it does, why is unsafe and why is needed to do it after having the instance?